src/extension/prelude.rs

```diff
@@ -71,7 +89,7 @@ pub const USIZE_T: Type = Type::new_extension(USIZE_CUSTOM_T);
 pub const BOOL_T: Type = Type::new_simple_predicate(2);
 
 /// Initialize a new array of type `typ` of length `size`
-pub fn new_array(typ: Type, size: u64) -> Type {
+pub fn array_type(typ: Type, size: u64) -> Type {
```

Comment:
driveby repeat of other nit (inc the comment on the line above), it's the *element* type